Kudu.Services/Diagnostics/ProfileManager.cs

```diff
@@ -0,0 +1,314 @@
+﻿using Kudu.Contracts.Tracing;
+using Kudu.Core.Infrastructure;
+using Kudu.Core.Tracing;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace Kudu.Services.Performance
+{
+    internal static class ProfileManager
+    {
+        private const int ProcessExitTimeoutInSeconds = 20;
+
+        private static ConcurrentDictionary<int, ProfileInfo> _profilingList = new ConcurrentDictionary<int, ProfileInfo>();
+        private static object _lockObject = new object();
+
+        // The profiling session timeout, this is temp fix before VS2015 Update 1.
+        private static readonly TimeSpan _profilingTimeout = TimeSpan.FromMinutes(15);
+        private static Timer _profilingIdleTimer;
+
+        private static string _processName = System.Environment.ExpandEnvironmentVariables("%SystemDrive%\\msvsmon\\profiler\\VSStandardCollector.Dev14.exe");
+
+        internal static async Task<HttpResponseInfo> StartProfileAsync(int processId, ITracer tracer = null)
```

Comment:
I don't think this class should return `HttpResponseInfo`, The class doesn't need to take any dependency on HTTP. It should return a value in success and throwing exceptions on errors.